### PR TITLE
macOS build fixes for thirdparty/qrcode

### PR DESCRIPTION
**Pull request by aaronzink**
only run sed on .c and .h files for zbar then delete the backup files
specify to use python2 for fiducial download script
update readmes

### CODE REVIEW

File: thirdparty/qrcode/zbar/build.py [hunk 1]
```diff
@@ -11,7 +11,8 @@
 
 # Build the library
 check_cd("zbar-0.10")
-run_command("find ./zbar -type f -print0 | xargs -0 sed -i 's/dprintf/zbarprintf/g'")
+run_command("find ./zbar -type f \(-iname '*.[ch]' \) -print0 | xargs -0 sed -i.bu 's/dprintf/zbarprintf/g'")
```
Comment (lessthanoptimal) [inline thirdparty/qrcode/zbar/build.py]: what was this change needed for?

Comment (aaronzink) [reply thirdparty/qrcode/zbar/build.py]: 1. Specifying ```-iname '*.[ch]'``` for find means that build.py will work if it is run a second time. This is useful if the first build failed due to a missing dependency etc. The reason build.py was failing on the second run was that it sees .o and other binary files in the ./zbar/zbar-0.10/zbar folder and tries to interpret them as text, resulting in ```sed: RE error: illegal byte sequence```.

2. sed on macOS gives and error like ```sed: 1: "./zbar/build.py": invalid command code .``` when using the -i option without specifying the extension. Another option is to use ```-i ''``` but I saw a [comment](https://stackoverflow.com/questions/7573368/in-place-edits-with-sed-on-os-x#comment47604810_7573438) saying that was not compatible with all versions of sed so I went with specifying the backup file extension and then deleting the backup files using find.

Comment (lessthanoptimal) [reply thirdparty/qrcode/zbar/build.py]: thanks for the clarification. testing on a linux box now

**Comment by lessthanoptimal** [comment]
successfully built zbar in linux
